### ISSUE TITLE
code 400,  type ModelException,   "Custom pip package installation failed for *my_model*"

### ISSUE THREAD

**Issue by Leoduv** [pytorch/serve#628]
## Context
Trying to get a model working with the necessary libraries in a requirements.txt added to the mar file following the documentation. 
If the container : 
* torchserve version: 0.2.0
* torch version: 1.5
* torchvision version [if any]: 0.6
* java version: 11 
* Operating System and version: Ubuntu 18.04.4 LTS

## Your Environment
* Installed using source? [yes/no]: yes, built with the following command ` DOCKER_BUILDKIT=1 docker build --file Dockerfile --build-arg BASE_IMAGE=nvidia/cuda:10.2-cudnn7-runtime-ubuntu18.04 -t torchserve-gpu:io`
* Deployed using docker container? [yes/no]: yes
* Is it a CPU or GPU environment?: gpu
* Using a default/custom handler? [If possible upload/share custom handler/model]: custom handler
* What kind of model is it e.g. vision, text, audio?: image segmentation 
* Provide config.properties, logs [ts.log] and parameters used for model registration/update APIs:

> ### ts_log.log
```
Torchserve version: 0.2.0
TS Home: /home/venv/lib/python3.6/site-packages
Current directory: /home/model-server
Temp directory: /home/model-server/tmp
Number of GPUs: 1
Number of CPUs: 2
Max heap size: 1866 M
Python executable: /home/venv/bin/python3
Config file: /home/model-server/config.properties
Inference address: http://0.0.0.0:8080
Management address: http://0.0.0.0:8081
Metrics address: http://127.0.0.1:8082
Model Store: /home/model-server/model-store
Initial Models: N/A
Log dir: /home/model-server/logs
Metrics dir: /home/model-server/logs
Netty threads: 32
Netty client threads: 0
Default workers per model: 1
Blacklist Regex: N/A
Maximum Response Size: 6553500
Maximum Request Size: 6553500
Prefer direct buffer: false
Custom python dependency for model allowed: true
Metrics report format: prometheus
Enable metrics API: true
2020-08-19 12:18:41,038 [INFO ] main org.pytorch.serve.ModelServer - Initialize Inference server with: EpollServerSocketChannel.
2020-08-19 12:18:41,165 [INFO ] main org.pytorch.serve.ModelServer - Inference API bind to: http://0.0.0.0:8080
2020-08-19 12:18:41,166 [INFO ] main org.pytorch.serve.ModelServer - Initialize Management server with: EpollServerSocketChannel.
2020-08-19 12:18:41,175 [INFO ] main org.pytorch.serve.ModelServer - Management API bind to: http://0.0.0.0:8081
2020-08-19 12:18:41,176 [INFO ] main org.pytorch.serve.ModelServer - Initialize Metrics server with: EpollServerSocketChannel.
2020-08-19 12:18:41,179 [INFO ] main org.pytorch.serve.ModelServer - Metrics API bind to: http://127.0.0.1:8082
2020-08-19 12:18:58,199 [INFO ] epollEventLoopGroup-3-1 org.pytorch.serve.archive.ModelArchive - eTag 1285dd4157d0483da09a09621713b06b
2020-08-19 12:18:58,215 [DEBUG] epollEventLoopGroup-3-1 org.pytorch.serve.wlm.ModelVersionedRefs - Adding new version 7.0 for model my_model
2020-08-19 12:18:58,215 [DEBUG] epollEventLoopGroup-3-1 org.pytorch.serve.wlm.ModelVersionedRefs - Setting default version to 7.0 for model my_model
2020-08-19 12:18:58,216 [INFO ] epollEventLoopGroup-3-1 org.pytorch.serve.wlm.ModelManager - Model my_model loaded.
```

> ### config.properties
```
inference_address=http://0.0.0.0:8080
management_address=http://0.0.0.0:8081
number_of_netty_threads=32
job_queue_size=1000
model_store=/home/model-server/model-store
install_py_dep_per_model=true
```

> ### model registration
`curl -X POST http://localhost:8081/models?url=my_model.mar&batch_size=2&initial_workers=1`

## Expected Behavior
If I understand correctly the packages in the requirements.txt file should be installed when a model is registered.

## Current Behavior

When trying to register, get response : `"code": 400,
  "type": "ModelException",
  "message": "Custom pip package installation failed for my_model"`
If it helps, I tried using `pip` inside the container and got `ImportError: cannot import name 'sysconfig'`

**Comment by harshbafna** [comment]
A similar error was observed in #577. Looks like the pip command crashes in the docker image created using this [dockerfile](https://github.com/pytorch/serve/blob/master/docker/Dockerfile)

**Comment by harshbafna** [comment]
@Leoduv: I have fixed the docker file in #577. Could you try using this new docker file?

**Comment by Leoduv** [comment]
Sorry for the wait, issue solved ! Thank you, I'll close the issue
